discovery/src/main/java/io/airlift/discovery/client/HttpDiscoveryAnnouncementClient.java

```diff
@@ -132,12 +133,21 @@ private static String getBodyForError(Response response)
         }
 
         Request request = prepareDelete()
-                .setUri(URI.create(uri + "/v1/announcement/" + nodeInfo.getNodeId()))
+                .setUri(buildAnnouncementUri(uri, nodeInfo.getNodeId()))
                 .setHeader("User-Agent", nodeInfo.getNodeId())
                 .build();
         return httpClient.executeAsync(request, new DiscoveryResponseHandler<>("Unannouncement", uri));
     }
 
+    private static URI buildAnnouncementUri(URI baseUri, String nodeId)
+    {
+        return HttpUriBuilder.uriBuilderFrom(baseUri)
+                .appendPath("v1")
+                .appendPath("announcement")
```

Comment:
Just `.appendPath("/v1/announcement")`, as we do eg here 
https://github.com/prestosql/presto/blob/118d1a1c65ed0eb5f5030edd8548900e23b092d1/presto-main/src/main/java/io/prestosql/server/remotetask/HttpLocationFactory.java#L54

Also, you can borrow method naming from there: `createAnnouncementLocation`.